payments-core/src/main/java/com/stripe/android/networking/StripeApiRepository.kt

```diff
@@ -320,6 +368,52 @@ internal class StripeApiRepository @JvmOverloads internal constructor(
         }
     }
 
+    /**
+     * Retrieve a [SetupIntent] using its client_secret, with the accepted payment method types
+     * ordered according to the [locale] provided.
+     *
+     * Analytics event: [AnalyticsEvent.SetupIntentRetrieve]
+     *
+     * @param clientSecret client_secret of the SetupIntent to retrieve
+     * @param locale locale used to determine the order of the payment method types
+     */
+    @Throws(
+        AuthenticationException::class,
+        InvalidRequestException::class,
+        APIConnectionException::class,
+        APIException::class
+    )
+    override suspend fun retrieveSetupIntentWithOrderedPaymentMethods(
+        clientSecret: String,
+        options: ApiRequest.Options,
+        locale: String
+    ): SetupIntent? {
+        fireFraudDetectionDataRequest()
+
+        val params = createClientSecretParam(
+            clientSecret,
+            listOf("setup_intent")
+        ).plus(
+            mapOf(
+                "type" to "setup_intent",
+                "locale" to locale
+            )
+        )
+
+        return fetchStripeModel(
+            apiRequestFactory.createPost(
+                getApiUrl("payment_method_preferences"),
+                options,
+                params
+            ),
+            PaymentMethodPreferenceForSetupIntentJsonParser()
+        ) {
+            fireAnalyticsRequest(
+                analyticsRequestFactory.createRequest(AnalyticsEvent.SetupIntentRetrieve)
+            )
+        }
```

Comment:
Even if externally we offer a Stripe and PaymentIntent option, internally would it make sense to have a function with the common code?